### PR TITLE
feat(dashboard): Recent People drawer is now in local storage

### PR DESCRIPTION
**Pull request by friedaxvictoria**
When one refreshes the page now the latest toggle state will reapear. 

do not merge before: #512

### CODE REVIEW

File: Phonebook.Frontend/src/app/pages/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -82,4 +99,23 @@ export class DashboardComponent implements OnInit {
   public removeFromBookmarkedPersons(person: Person) {
     this.store.dispatch(new ToggleBookmark(person));
   }
+
+  public setDrawer() {
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/pages/dashboard/dashboard.component.ts]: Please use NGXS as state manager. (https://www.ngxs.io/) 
You can add it to https://github.com/T-Systems-MMS/phonebook/blob/master/Phonebook.Frontend/src/app/shared/states/App.state.ts

Comment (friedaxvictoria) [reply Phonebook.Frontend/src/app/pages/dashboard/dashboard.component.ts]: Do I have to add a service in order to do that?

**Comment by friedaxvictoria** [comment]
The state of the drawer now uses a state manager. Currently the drawer doesn't close when the window reaches a certain size. Though one can still close the window manually. Should I try and change that again?

**Comment by DanielHabenicht** [comment]
> The state of the drawer now uses a state manager. Currently the drawer doesn't close when the window reaches a certain size. Though one can still close the window manually. Should I try and change that again?

Well, that's the question that was creeping up for me already. If we want the user to control the drawer and keep the state. There needs to be some logic deciding if it should auto open. Maybe we should safe the state according to the screen sizes, meaning: 
Scaling from small to big screen: 
If the user hid/showed the recent people drawer on big screen do hide/show it if it rescales. 
Scaling from big to small screen:
Simply ignore the state and show/hide it acccording to the breakpoint

**Comment by friedaxvictoria** [comment]
@DanielHabenicht The last commit actually works the way you implemented it in the first place: You can hide/show it whenever and wherever you want. When you scale it from a bigger to a smaller screen the current state stays active. However, when reloaded, the drawer automatically stays hidden. So it only works on refreshes but that's the way it worked before as well.

**Comment by DanielHabenicht** [comment]
Well, the implementation on the other pr worked also while resizing the window. 
In short, I think the state should only be persisted for the big screen view. While maintaining auto hiding/showing the drawer for different widths.

**Comment by friedaxvictoria** [comment]
When I open the preview for the other pr it doesn't work while resizing. It's only automatically hidden when I refresh the page. I can try and implement it in this version but it still won't work in the other pr.

**Comment by friedaxvictoria** [comment]
Now the drawer opens automatically when the window is wider than 768px and closes automatically when it's smaller. I don't know whether the animation/movement is smooth enough though. It also doesn't use the breakpointObserver anymore. I left it as a comment so it's easily restored in case that version was better after all. 

Should we open another issue to auto hide the drawer in the room-tree-component?

**Comment by friedaxvictoria** [comment]
It doesn't work in the demo...

**Comment by friedaxvictoria** [comment]
Now it works just fine.

**Comment by DanielHabenicht** [comment]
Nice, this looks good, But for: 
big screen -> small screen --> big screen 
the preference is not saved. if it was hidden before it is now expanded.

**Comment by friedaxvictoria** [comment]
Yeah, I did that on purpose but if it's better the other way I'll change it.

File: Phonebook.Frontend/src/app/pages/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -24,10 +40,33 @@ export class DashboardComponent implements OnInit {
...
+  constructor(private store: Store, private cd: ChangeDetectorRef, private breakpointObserver: BreakpointObserver) {}
 
   public ngOnInit() {
     this.changeOrder();
+    const activeDrawerState = this.store.selectOnce(state => state.appstate.activeDrawer);
+    activeDrawerState.subscribe(d => {
+      this.activeDrawer = d;
+      if (this.activeDrawer) {
+        this.drawerOpened = true;
+      }//if (this.drawerOpened) {
```
Comment (paule96) [inline Phonebook.Frontend/src/app/pages/dashboard/dashboard.component.ts]: Can we remove the commented things?

Comment (friedaxvictoria) [reply Phonebook.Frontend/src/app/pages/dashboard/dashboard.component.ts]: Yes

**Comment by DanielHabenicht** [comment]
@paule96 @friedaxvictoria @Zeroks77 bitte nochmal drüberschauen und gucken ob es mit dem resizing jetzt so funktioniert wie ihr es erwarten würdet. 
Er merkt sich den Status nur solange der Bildschirm groß ist. Ansonsten ist fährt der recentPeopleDrawer standardmäßig ein.

**Comment by friedaxvictoria** [comment]
Ich finde, dass es gut aussieht!

**Comment by paule96** [comment]
@mmsgithub-ci /help

**Comment by DanielHabenicht** [comment]
Kann der Bot etwas was ich noch nicht weiß? @paule96

**Comment by paule96** [comment]
> Kann der Bot etwas was ich noch nicht weiß? @paule96

Ich hatte gehofft das er das kann. Ich will wollte doch nur wissen wie ich die pipeline nochmal anschubse

**Comment by DanielHabenicht** [comment]
Dann kannste ja jetzt deinen haken ransetzen ;) @paule96

**Pull request merged**